libraries/extractor/src/main/java/androidx/media3/extractor/ts/DtsUhdReader.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (C) 2016 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package androidx.media3.extractor.ts;
+
+import static java.lang.Math.min;
+
+import androidx.annotation.Nullable;
+import androidx.media3.common.C;
+import androidx.media3.common.Format;
+import androidx.media3.common.ParserException;
+import androidx.media3.common.util.Assertions;
+import androidx.media3.common.util.ParsableByteArray;
+import androidx.media3.common.util.UnstableApi;
+import androidx.media3.common.util.Util;
+import androidx.media3.extractor.DtsUtil;
+import androidx.media3.extractor.ExtractorOutput;
+import androidx.media3.extractor.TrackOutput;
+import androidx.media3.extractor.ts.TsPayloadReader.TrackIdGenerator;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+import org.checkerframework.checker.nullness.qual.RequiresNonNull;
+
+/** Parses a continuous DTS UHD byte stream and extracts individual samples. */
+@UnstableApi
+public final class DtsUhdReader implements ElementaryStreamReader {
```

Comment:
Is there a specific reason to declare a new class for UHD? Can we not accommodate  it in `DtsReader`? As far as I can tell from the implementation we can add variables specific to UHD into DtsReader as we have done for HD.

Reply:
@rohitjoins, Please let me know if any further code updates are required on this CL. Thank you.

Reply:
Is there any update?

Reply:
@rohitjoins , would you please suggest further action required to merge these changes?

Reply:
I think it would be better to have a single DTS reader class to handle all types of DTS stream, we won't have to make the same changes in two different places as most of the logic is repeated.

Reply:
> I think it would be better to have a single DTS reader class to handle all types of DTS stream, we won't have to make the same changes in two different places as most of the logic is repeated.

This is done in the commit [ef41526](https://github.com/androidx/media/pull/275/commits/ef415268e215719ce385f5fb0e3149679a3bd91e)